torchvision/utils.py

```diff
@@ -363,24 +377,43 @@ def draw_keypoints(
     elif image.size()[0] != 3:
         raise ValueError("Pass an RGB image. Other Image formats are not supported")
 
+    # validate keypoints
     if keypoints.ndim != 3:
         raise ValueError("keypoints must be of shape (num_instances, K, 2)")
 
+    # validate visibility
+    if visibility is None:  # set default
+        visibility = torch.ones(keypoints.shape[:-1], dtype=torch.bool)
+    # If the last dimension is 1, e.g., after calling split([2, 1], dim=-1) on the output of a keypoint-prediction
+    # model, make sure visibility has shape (num_instances, K).
+    # Iff K = 1, this has unwanted behavior, but K=1 does not really make sense in the first place.
+    visibility.squeeze_(-1)
+    if visibility.ndim != 2:
+        raise ValueError("visibility must be of shape (num_instances, K)")
```

Comment:
Let's provide some debugging info as well:

```suggestion
        raise ValueError(f"visibility must be of shape (num_instances, K). Got ndim={visibility.ndim}")
```